reconcile/database_access_manager.py

```diff
@@ -0,0 +1,506 @@
+                    for resource in external_resource.resources:
+                        engine = get_db_engine(resource)
+                        for db_access in resource.database_access:
+                            resource_name = f"dbam-{db_access.name}"
+                            oc = oc_map.get_cluster(namespace.cluster.name, False)
+
+                            managed_resources = self._populate_resources(
+                                db_access,
+                                engine,
+                                image_repository,
+                                pull_secret,
+                                resource,
+                                resource_name,
+                                settings,
+                            )
+
+                            # create job, delete old, failed job first
+                            job = oc.get(
+                                namespace.name,
+                                "Job",
+                                f"dbam-{db_access.name}",
+                                allow_not_found=True,
+                            )
+                            if job:
+                                conditions = [
+                                    JobStatusCondition(type=c["type"])
+                                    for c in job["status"].get("conditions", [])
+                                ]
+                                job_status = JobStatus(conditions=conditions)
+                                if job_status.is_complete():
+                                    # if job_status.has_errors():
+                                    #     logging.info(
+                                    #         f"Deleting previous failed job job dbam-{db_access.name}"
+                                    #     )
+                                    oc.delete(
+                                        namespace.name,
+                                        "Job",
+                                        f"dbam-{db_access.name}",
+                                    )
+                                else:
+                                    logging.error(
+                                        f"Job dbam-{db_access.name} appears to be still running"
+                                    )
+                                    continue
+
+                            for r in managed_resources:
+                                openshift_base.apply(
+                                    dry_run=dry_run,
+                                    oc_map=oc_map,
+                                    cluster=namespace.cluster.name,
+                                    namespace=namespace.name,
+                                    resource_type=r.resource.kind,
+                                    resource=r.resource,
+                                    wait_for_namespace=False,
+                                )
+
+                            while True:
```

Comment:
so the integration creates a job and waits for the results. what happens if the integration gets killed in between (various reasons from cluster upgrades to qr promotion).

Reply:
sgtm. but why the while true then? you can pick up where you left of during the next reconcile run, no?